client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js

```diff
@@ -120,6 +128,82 @@ export class ShippingBanner extends Component {
 		} );
 	};
 
+	async acceptTosAndGetWCSAssets() {
+		const { acceptTos, getWcsAssets } = this.props;
+
+		const accepted = await acceptTos();
+
+		if ( accepted ) {
+			const wcsAssets = await getWcsAssets();
+			this.loadWcsAssets( wcsAssets );
+		}
+	}
+
+	loadWcsAssets( { assets } ) {
+		if ( this.state.wcsAssetsLoaded || this.state.wcsAssetsLoading ) {
+			this.openWcsModal();
+			return;
+		}
+
+		this.setState( { wcsAssetsLoading: true } );
+
+		const js = assets.wc_connect_admin_script;
+		const styles = assets.wc_connect_admin_style;
+
+		const shippingLabelContainer = document.createElement( 'div' );
+		const { orderId } = this.state;
+
+		shippingLabelContainer.className =
+			'wcc-root woocommerce wc-connect-create-shipping-label';
+		shippingLabelContainer.dataset.args = JSON.stringify( {
+			orderId,
+			context: 'shipping_label',
+		} );
+		document.body.appendChild( shippingLabelContainer );
+
+		Promise.all( [
+			new Promise( ( resolve, reject ) => {
+				const script = document.createElement( 'script' );
+				script.src = js;
+				script.async = true;
+				script.onload = resolve;
+				script.onerror = reject; //TODO: handle errors
+				document.body.appendChild( script );
+			} ),
+			new Promise( ( resolve, reject ) => {
+				const head = document.getElementsByTagName( 'head' )[ 0 ];
+				const link = document.createElement( 'link' );
+				link.rel = 'stylesheet';
+				link.type = 'text/css';
+				link.href = styles;
+				link.media = 'all';
+				link.onload = resolve;
+				link.onerror = reject; //TODO: handle errors
```

Comment:
Let's make sure we mention this line in our error-handling issue, @fer, so we can clean this comment up